src/app/core/services/repo-creator.service.ts

```diff
@@ -2,21 +2,53 @@ import { Injectable } from '@angular/core';
 import { flatMap } from 'rxjs/operators';
 import { Observable, of, pipe, UnaryFunction } from 'rxjs';
 import { GithubService } from './github.service';
+import { UserService } from './user.service';
+import { Phase } from '../models/phase.model';
+import { UserRole } from '../models/user.model';
+
+export const MISSING_REQUIRED_REPO = 'You cannot proceed without the required repository.';
+export const CURRENT_PHASE_REPO_CLOSED = 'Current Phase\'s Repository has not been opened.';
+export const BUG_REPORTING_INVALID_ROLE =
+  "'Bug-Reporting Phase\'s repository initialisation is only available to Students.'";
+
 
 @Injectable({
   providedIn: 'root',
 })
 export class RepoCreatorService {
   constructor(
-    private githubService: GithubService
+    private githubService: GithubService,
+    private userService: UserService
   ) {}
 
+  /**
+   * Attempts to create the repository if permissions have been given to do so.
+   * @param currentPhase the current phase of the session.
+   * @param phaseRepo the name of the specified repository.
+   */
+  public attemptRepoCreation(currentPhase: Phase, phaseRepo: string):
+    UnaryFunction<Observable<boolean | null>, Observable<boolean | null>> {
+    return pipe(
+      flatMap((sessionFixPermission: boolean | null) => {
```

Comment:
```suggestion
      flatMap((repoCreationPermission: boolean | null) => {
```

Let's rename the variable for a more obvious meaning

Reply:
Aights!